be/src/storage/tablet_schema.cpp

```diff
@@ -546,6 +662,48 @@ void TabletSchema::to_schema_pb(TabletSchemaPB* tablet_schema_pb) const {
     tablet_schema_pb->mutable_sort_key_idxes()->Add(_sort_key_idxes.begin(), _sort_key_idxes.end());
     tablet_schema_pb->mutable_sort_key_unique_ids()->Add(_sort_key_uids.begin(), _sort_key_uids.end());
     tablet_schema_pb->set_schema_version(_schema_version);
+    for (auto& index : _indexes) {
+        auto* tablet_index_pb = tablet_schema_pb->add_table_indices();
+        index.to_schema_pb(tablet_index_pb);
+    }
+}
+
+Status TabletSchema::get_indexes_for_column(int32_t col_unique_id,
+                                            std::unordered_map<IndexType, TabletIndex>* res) const {
+    RETURN_IF(res != nullptr, Status::InternalError("Index map should not be nullptr"));
+    for (const auto& _index : _indexes) {
+        if (_index.col_unique_ids().size() == 1) {
+            for (int32_t id : _index.col_unique_ids()) {
```

Comment:
It is confusing to use for loop, because we can confirm that the size is 1.
```suggestion
            if (_index.col_unique_ids()[0] == col_unique_id) { {
```